### PR TITLE
Project: Fix NextJS ESLint build warnings

### PR DESCRIPTION
**Pull request by eatyourgreens**
- configure ESLint for NextJS Core Web Vitals.
- ignore `.stories.js` files.
- refactor `WorkflowAssignmentModal` to replace `useEffect` with `useMemo`.
- refactor `AboutNavLink` to remove effect hook.
- fix `useEffect` in `RecentSubjectsContainer`.
- fix `useEffect` in `_app.js`.
- fix anonymous default exports in `localeMenu` and `AuthModal` theme.

Note that the build warnings for `SubjectPicker` are already fixed in #3585.

## Package
app-project

## How to Review
Run `yarn build` for the project app. You should see only a handful of warnings now, mostly about replacing `` with `next/image` for static images.

The app should start and run as usual, without any changes.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by eatyourgreens** [comment]
The state logic in `WorkflowAssignmentModalContainer` is a little hard to follow, but tests are passing so I don't think I've broken it.

For clarity, and future maintainers of that code, it might be useful to rewrite that modal following [the new React guidelines for derived state](https://beta.reactjs.org/apis/react/useState).

**Review by goplayoutside3 — COMMENTED** [review]
There are a couple bugs with defining a placeholder image in this PR. I recommend creating a storybook story for cases where a placeholder image is expected.

Team member in i-fancy-cats:
team

Recently classified subjects in any project where I haven't made a classification:
recents

**Comment by eatyourgreens** [comment]
D'oh! Those are actually bugs on master. The asset prefix isn't defined in development! That must have happened in #3600.

**Comment by eatyourgreens** [comment]
I’ve opened #3698 for the broken images.

**Comment by eatyourgreens** [comment]
Placeholders are also broken in the content pages app.

**Comment by eatyourgreens** [comment]
The [story for team member placeholders](https://zooniverse.github.io/front-end-monorepo/@zooniverse/fe-project/index.html?path=/story/project-app-screens-about-pages-teammember--placeholder&globals=locale:en;locales.en:English;locales.test:Test+Language) is broken too.

**Comment by eatyourgreens** [comment]
I've added development asset prefixes to both apps, to fix the build warning and close #3698.

**Comment by goplayoutside3** [comment]
> I've also added Recent Subjects stories with placeholders. Those will be broken since the Storybook doesn't run NextJS, and the placeholders are served from `/_next`.

I don't think this is a blocking question, but why does TeamMember's placeholder display correctly in Storybook, but RecentSubject's placeholder does not? It seems like both components are using them same lines of code for getting `assetPrefix`.

**Comment by eatyourgreens** [comment]
> I don't think this is a blocking question, but why does TeamMember's placeholder display correctly in Storybook, but RecentSubject's placeholder does not? It seems like both components are using them same lines of code for getting assetPrefix.

I haven't got the storybook open in front of me, but I think that if you right click and open each placeholder in a new tab, the missing image is using `next/image`, which loads images from `/_next`.

**Comment by eatyourgreens** [comment]
The fully qualified URL for that subject placeholder is https://www.zooniverse.org/projects/_next/image?url=%2Fprojects%2Fassets%2Fsubject-placeholder.png&w=1080&q=75, which does work for me. I'm not quite sure how to get the story to use that URL.

**Comment by eatyourgreens** [comment]
Maybe building the storybook with `APP_ENV=production` would force it to use production URLs for images. That might work.

**Comment by goplayoutside3** [comment]
That might work - I think I was more curious about the difference between TeamMember and RecentSubjects. Their placeholder code looks identical to me, but for some reason one works in storybook and the other does not.

**Comment by eatyourgreens** [comment]
NextJS was complaining that `RecentSubjects` should use `<Image>` instead of ``, so I changed that here to remove the build warning. It seems fine with the `TeamMember` placeholder though. No warnings.

**Comment by eatyourgreens** [comment]
`TeamMember` is using Grommet `Image` but `RecentSubjects` is using NextJS `Image`. Different components with the same name.

**Comment by goplayoutside3** [comment]
> `TeamMember` is using Grommet `Image` but `RecentSubjects` is using NextJS `Image`. Different components with the same name.

I see, thanks for the clarification!

**Review by goplayoutside3 — APPROVED** [review]
